chia/wallet/wallet_node.py

```diff
@@ -929,16 +941,20 @@ async def get_timestamp_for_height(self, height: uint32) -> uint64:
             if cache_ts is not None:
                 return cache_ts
 
-        peer: Optional[WSChiaConnection] = self.get_full_node_peer()
-        if peer is None:
-            raise ValueError("Cannot fetch timestamp, no peers")
-        self.log.debug(f"Fetching block at height: {height}")
-        last_tx_block: Optional[HeaderBlock] = await fetch_last_tx_from_peer(height, peer)
+        peers: List[WSChiaConnection] = self.get_full_node_peers_in_order()
+        last_tx_block: Optional[HeaderBlock] = None
+        for peer in peers:
+            self.log.debug(f"Fetching block at height: {height} from {peer.get_peer_info()}")
+            last_tx_block = await fetch_last_tx_from_peer(height, peer)
+            if last_tx_block is None:
+                continue
+
+            assert last_tx_block.foliage_transaction_block is not None
+            self.get_cache_for_peer(peer).add_to_blocks(last_tx_block)
+            return last_tx_block.foliage_transaction_block.timestamp
+
         if last_tx_block is None:
```

Comment:
wouldn't `last_tx_block` always be None here?

Reply:
fixed